src/providers/asciidoc.provider.ts

```diff
@@ -0,0 +1,97 @@
+import * as vscode from "vscode";
+import * as path from "path";
+import { createContext, Context } from "./createContext";
+
+import { createPathCompletionItem } from "./createCompletionItem";
+import {
+  getPathOfFolderToLookupFiles,
+  getChildrenOfPath,
+  sortFilesAndDirectories,
+} from "../util/file";
+
+export const AsciidocProvider = {
+  provideCompletionItems,
+};
+
+export async function provideCompletionItems(
+  document: vscode.TextDocument,
+  position: vscode.Position
+): Promise<vscode.CompletionItem[]> {
+  const context = createContext(document, position);
+
+  return shouldProvide(context)
+    ? provide(context)
+    : Promise.resolve([]);
+}
+
+/**
+ * Checks if we should provide any CompletionItems
+ * @param context
+ */
+function shouldProvide(context: Context): boolean {
+  return /(include\:\:|image\:\:|image\:)\S*/gi.test(context.textFullLine)
+}
+
+/**
+ * Provide Completion Items
+ */
+async function provide(
+  context: Context
+): Promise<vscode.CompletionItem[]> {
+  const documentText = context.document.getText();
+  const pathExtractedFromIncludeString = context.textFullLine.replace('include::', '').replace('image::', '').replace('image:', '');
```

Comment:
Ideally we would resolve `image::` references in the completion relative to the document base directory and the `imagesdir` if set.